### ISSUE TITLE
Emoji - Inconsistency between Android and mWeb emojis are not highlighted on android during search

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#42762]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.77-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** PR passed?
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/42407
## Action Performed:
1. Open the app and log in
2. Open any report
3. Tap the emoji picker
4. Enter a search word that would return only one result and not in recently used, e.g. panda, fox, pizza etc
5. Notice emoji not highlighted on android

## Expected Result:
Emoji should be highlighted on Android

## Actual Result:
Inconsistency between Android and mWeb emojis are not highlighted on android during search

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/acff61f8-2edd-4fe3-9c43-959e4cf4000d

https://github.com/Expensify/App/assets/78819774/0ec654ef-be29-4c1a-9b7a-c0efd6654924

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
There is an inconsistency on how we highlight the first emoji when searching.

### What is the root cause of that problem?
On web, we have a logic to highlight the first emoji when searching, however, this logic doesn't present for native.
https://github.com/Expensify/App/blob/e177a6974596dcca88fb84ba7686b99dedace271/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L262
https://github.com/Expensify/App/blob/e177a6974596dcca88fb84ba7686b99dedace271/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L283

The way it works is, if we are searching, we set `highlightFirstEmoji` to true,
https://github.com/Expensify/App/blob/e177a6974596dcca88fb84ba7686b99dedace271/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L127

we reset it whenever the user uses keyboard to move to other emojis,
https://github.com/Expensify/App/blob/e177a6974596dcca88fb84ba7686b99dedace271/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L78-L80

or when the user hover over an emoji,
https://github.com/Expensify/App/blob/e177a6974596dcca88fb84ba7686b99dedace271/src/components/EmojiPicker/EmojiPickerMenu/index.tsx#L272-L274

both of these things don't exist in native.

### What changes do you think we should make in order to solve the problem?

We can just copy the logic from web file to native file. We set highlightFirstEmoji when searching and clear it otherwise. We don't need to clear it when hovering or using keyboard key because both things don't exist in native.

Or, we can just the highlight condition to
```
const shouldFirstEmojiBeHighlighted = index === 0;
```

Why? When we are not searching, we are showing a Frequently used header. The header takes the 0 to 7 index, so the first emoji index will be 8. If we are searching, the header is hidden, thus the first emoji index will be 0.

**Comment by bernhardoj** [comment]
@techievivek @ishpaul777 While working on the PR, I'm questioning whether we should do this or not. In web, we can use keyboard to navigate and select the emoji, so it makes sense to highlight the first emoji so the user can press Enter to select it, but on native, we don't support keyboard control, so even if we highlight it, pressing Enter will do nothing. Do we still want to do it?

**Comment by lschurr** [comment]
@techievivek could you take a look here?

**Comment by techievivek** [comment]
Yeah, I don't know why it needs to be highlighted. It makes sense for web but not for mobile. Let's confirm with design team on this behaviour. Thanks.,

**Comment by dannymcclain** [comment]
I think we highlight the emoji on desktop because you can use your keyboard (`return/enter`) to add it to the composer. That's not really as common of a pattern on mobile—I mean I guess technically we could implement it that way... but it's just not as common on mobile devices to use the return key to do things like that. So I think we can leave this behavior as is and close this.

**Comment by techievivek** [comment]
Great, thanks for confirming. We were on same boat so closing this out.
